### PR TITLE
Fb1220

### PR DESCRIPTION
**Pull request by spireken**
I created a new folder on Dropbox and put the build in it.

### CODE REVIEW

File: Source/Ashkal/main.cpp [hunk 1]
```diff
@@ -1,13 +1,75 @@
...
+#include "Ashkal/Shapes/Shape.hpp"
+#include "Ashkal/Shapes/Stage.hpp"
 
 using namespace Ashkal;
 
+class Viewport : public QWidget {
+  public:
+
+    Viewport(const Stage& stage)
+      : m_stage(&stage) {
```
Comment (kranar) [inline Source/Ashkal/main.cpp]: Double indent line 18.

File: Source/Shapes/Circle.cpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#include "Ashkal/Shapes/Circle.hpp"
+
+using namespace Ashkal;
+
+Circle::Circle() {
```
Comment (kranar) [inline Source/Shapes/Circle.cpp]: Use a delegating constructor:

https://en.cppreference.com/w/cpp/language/initializer_list

File: Source/Shapes/Line.cpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#include "Ashkal/Shapes/Line.hpp"
+#include <climits>
+
+using namespace Ashkal;
+
+Line::Line() {
+  Point point;
```
Comment (kranar) [inline Source/Shapes/Line.cpp]: Use a delegating constructor:

https://en.cppreference.com/w/cpp/language/initializer_list

Initialize the Point passed into the delegated constructor using aggregate initialization:

https://en.cppreference.com/w/cpp/language/aggregate_initialization

File: Source/Shapes/Shape.cpp [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#include "Ashkal/Shapes/Shape.hpp"
+
+using namespace Ashkal;
+
+Shape::~Shape() {}
```
Comment (kranar) [inline Source/Shapes/Shape.cpp]: Eliminate this file.

Use an explicitly defaulted destructor:

https://en.cppreference.com/w/cpp/language/member_functions#Special_member_functions

**Review by kranar — CHANGES_REQUESTED** [review]

**Comment by spireken** [comment]
I forgot to add the Stage source files in the original commit, added them now.

File: Source/Shapes/Line.cpp [hunk 5]
```diff
@@ -3,12 +3,8 @@
...
 using namespace Ashkal;
 
-Line::Line() {
-  Point point;
-  point.x = 0;
-  point.y = 0;
-  m_point = point;
-}
+Line::Line()
+  : Line({ 0, 0 }) {}
```
Comment (kranar) [inline Source/Shapes/Line.cpp]: ```{0, 0}```

**Review by kranar — CHANGES_REQUESTED** [review]

**Pull request merged**